splicemachine/spliceengine
# DB-11539/DB-11784 EXPORT fixes

## Short Description
This changes does
- DB-11784: Add feature that IMPORT_DATA can now read from parquet
- DB-11539: fix EXPORT so that EXPORT uses DEFAULT_TIMESTAMP_FORMAT if timestamp_format is not set

## Long Description
- DB-11784: We can already EXPORT to parquet, now we can also IMPORT. Since we can import from a directory, we will use PARQUET import once we have one `.parquet` file in the directory. If we have .orc, we use orc. Otherwise, we use .csv .
- DB-11539: Before this, a change in global database property `splice.function.timestampFormat` would change the behavior of EXPORT when not timestamp_format was given and the "default" value should be taken.  Since only the EXPORT would use that default value, but not IMPORT, we had a mismatch between these two, so we aligned now both so that default EXPORT can be read by default EXPORT.
- Furthermore, this changes does some refactoring and improves the Export test to test that all EXPORTd files can be IMPORTd with the same settings.

## How to test DB-11784 IMPORT_DATA parquet

```
EXPORT TO '/tmp/parquet_test' AS PARQUET  VALUES (1, 'a'), (2, 'b'), (3, 'c');
call SYSCS_UTIL.IMPORT_DATA('SPLICE', 'PARQUET_TEST', null,
       '/tmp/parquet_test/', null, null, null, null, null, '0', '/tmp', false, null);
CREATE TABLE PARQUET_TEST ( I INTEGER, C VARCHAR(32));
SELECT * FROM PARQUET_TEST;
```
should result in
```
I          |C                               
--------------------------------------------
1          |a                               
2          |b                               
3          |c       
```

## How to test DB-11539 EXPORT should use DEFAULT_TIMESTAMP_FORMAT
The following code should run independent of setting of
`call syscs_util.syscs_set_global_database_property('splice.function.timestampFormat', X)`, where X can be `null`, `'true'` or `'false'`.

```
CREATE TABLE A ( T TIMESTAMP);
INSERT INTO A VALUES ('2020-01-01 12:15:16.123456789');
EXPORT TO '/tmp/timestamp_test' AS CSV SELECT * FROM A;

DELETE FROM A; -- reset table A
call SYSCS_UTIL.IMPORT_DATA('SPLICE', 'A', null, '/tmp/timestamp_test/', null,
     null, null, null, null, '0', '/tmp', false, null);
SELECT * FROM A;
-- 2020-01-01 12:15:16.123456   
```
